Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreState.java

```diff
@@ -46,6 +60,34 @@
      * An array of reference counts for each stored point.
      */
     private short[] refCount;
+    /**
+     * is direct mapping enabled
+     */
+    @Getter
+    private boolean direcMapLocation;
+    /**
+     * location data for indirect maps
+     */
+    private int[] locationList;
+    /**
+     * reverse location data to be usable in future
+     */
+    private int[] reverseLocationList;
+    /**
+     * flag to avoid null issues in the future
+     */
+    @Getter
+    private boolean reverseAvailable;
```

Comment:
Is this flag to solve the problem with null fields in protostuff? I think that's something that we'll want to handle in a consistent way in all our state classes.

Reply:
Yes. Looking ahead when the reverse array will be used. The nodeStore/LeafStore do not need these flags because they branch on the value of the capacity field. This may be relevant for the samplers.